src/GraphQL.ApiTests/GraphQL.approved.txt

```diff
@@ -501,8 +490,11 @@ namespace GraphQL.Builders
         public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Directive(string name, System.Action<GraphQL.Types.AppliedDirective> configure) { }
         public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Directive(string name, string argumentName, object argumentValue) { }
         public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Name(string name) { }
-        public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Resolve(GraphQL.Resolvers.IFieldResolver resolver) { }
+        public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Resolve(GraphQL.Resolvers.IFieldResolver<TReturnType> resolver) { }
         public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> Resolve(System.Func<GraphQL.IResolveFieldContext<TSourceType>, TReturnType> resolve) { }
+        public virtual GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> ResolveAsync(GraphQL.Resolvers.IFieldResolver<GraphQL.DataLoader.IDataLoaderResult<TReturnType>> resolver) { }
```

Comment:
It is important not to have `Async` in the method name here.  An asynchronous method returning a `Task<IDataLoaderResult>` is distinctly different than a synchronous method returning a `IDataLoaderResult`.  This is one of the bugs that people had been complaining about that was fixed in the new data loader design.  See #945 .